### PR TITLE
Boards page layout

### PR DESCRIPTION
**Pull request by malinka775**
add:
- add basic layout of boards page
- sending updates to server when create / delete board
- sending updates to server when create / reorder columns

screenshot:
Снимок экрана 2022-12-04 в 16 55 43

Снимок экрана 2022-12-04 в 16 56 51

### CODE REVIEW

File: src/layouts/Board/Board.tsx [hunk 1]
```diff
@@ -6,318 +6,113 @@ import BoardColumn from './BoardColumn';
...
-      order: columns.length,
-      boardId,
-      tasks: [],
-    };
-    const newColumns = [...columns];
-    newColumns.push(column);
-    setColumns(newColumns);
-    return column;
+  const addColumn = async (title: string) => {
+    console.log('adding');
```
[the remark this answers is not in the document]
Comment (malinka775) [reply src/layouts/Board/Board.tsx]: Removed

File: src/layouts/Boards/Boards.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      } else {
+        console.log('board removed successfully');
+        const newBoards = [...boards].filter((board) => board._id !== id);
+        setBoards(newBoards);
+      }
+    });
+  };
+
+  const goToBoardPage = (id: string) => {
+    console.log(id);
```
[the remark this answers is not in the document]
Comment (malinka775) [reply src/layouts/Boards/Boards.tsx]: removed

File: src/layouts/Boards/Boards.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    navigate(`/board/${id}`);
```
[the remark this answers is not in the document]
Comment (malinka775) [reply src/layouts/Boards/Boards.tsx:48]: Done

File: src/layouts/Boards/BoardCard/BoardCard.tsx [hunk 4]
```diff
@@ -10,44 +10,79 @@ import { isLargeMobile } from '../../../helpers/getWindowWidth';
...
+    const { children, className } = props;
+    const cardClassNames = classNames(classes.boardCard, className);
+    return (
+      <Card className={cardClassNames}>
+        <Card.Body className={classes.boardCard__content}>
+          <Card.Text as="div" className={classes.boardCard__info}>
+            {children}
+          </Card.Text>
+        </Card.Body>
+      </Card>
```
[the remark this answers is not in the document]
Comment (malinka775) [reply src/layouts/Boards/BoardCard/BoardCard.tsx:28]: Yes, so it takes place of 1 full boardCard...

File: src/layouts/Boards/Boards.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    });
+
+  const showModal = () => {
+    setIsModalActive(true);
+  };
+
+  const closeModal = () => {
+    setIsModalActive(false);
+  };
+
```
[the remark this answers is not in the document]
Comment (malinka775) [reply src/layouts/Boards/Boards.tsx:77]: Next PR

File: src/layouts/Board/Board.tsx [hunk 6]
```diff
@@ -6,318 +6,113 @@ import BoardColumn from './BoardColumn';
...
-      <div className={classes.board}>
-        <h1 className={classes.board__title}>{board?.title}</h1>
-        <div className={classes.board__content}>
-          <DragDropContext onDragEnd={onDragEnd}>
-            <BoardColumnsWrapper addColumn={addColumn}>
-              {renderColumns.length > 0 && renderColumns}
-            </BoardColumnsWrapper>
-          </DragDropContext>
+    <>
+      {isLoading && <Spinner></Spinner>}
```
[the remark this answers is not in the document]
Comment (malinka775) [reply src/layouts/Board/Board.tsx]: Refactored

File: src/layouts/Boards/BoardCard/BoardCard.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (malinka775) [reply src/layouts/Boards/BoardCard/BoardCard.tsx:28]: Maybe I could wrap is only with Card though

**Review by kravchuk-st — APPROVED** [review]

**Pull request merged**